src/lexer.rs

```diff
@@ -22,11 +22,9 @@ impl Lexer {
             pos: 0,
             line_count: 1,
         };
-        l.line = match l.lines.next() {
-            Some(line) => Some(line.unwrap()),
-            None => None,
-        };
-        return l;
+        l.line = l.lines.next().and_then(|l| l.ok());
```

Comment:
matches only doing something on the `Some` branch can be simplified using `.map` as `Option` implements iter